include/cocaine/rpc/dispatch.hpp

```diff
@@ -153,6 +155,11 @@ struct select<streamed<R>, Event, ForwardMeta> {
     typedef io::deferred_slot<streamed, Event, ForwardMeta> type;
 };
 
+template<class Event, class ForwardMeta>
+struct select<typename io::basic_slot<Event>::result_type, Event, ForwardMeta> {
```

Comment:
Why not `struct select<boost::optional<std::shared_ptr<const dispatch_type>>, Event, ForwardMeta`?